### PR TITLE
fix: Remove panic when we divide 0/0 in quotient directive

### PR DESCRIPTION
**Pull request by kevaundray**
# Description

A directive should ideally not panic when we are dividing zero by zero. The input is not trusted and should be constrained, so it is okay to return 0 when we have 0/0.

Related: https://github.com/noir-lang/noir/issues/1802

## Problem\*

Resolves 

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: acvm/src/pwg/directives/mod.rs [hunk 1]
```diff
@@ -58,7 +58,7 @@ fn solve_directives_internal(
                 None => FieldElement::one(),
             };
 
-            let (int_r, int_q) = if pred_value.is_zero() {
+            let (int_r, int_q) = if pred_value.is_zero() || int_b.is_zero() {
```
Comment (vezenovm) [inline acvm/src/pwg/directives/mod.rs:61]:
```suggestion
            let (int_r, int_q) = if pred_value.is_zero() || (int_a.is_zero() && int_b.is_zero()) {
```
Don't we want this? This will give us (0, 0) as the remainder and quotient for whenever int_b is zero. We should add a test for when divisor_is_zero but the dividend is not zero.

**Review by vezenovm — APPROVED** [review]
